torch/jit/frontend.py

```diff
@@ -226,21 +238,22 @@ def build_While(ctx, stmt):
             # annotations in this case
             raise NotSupportedError(None, "else branches of while loops aren't supported")
         r = ctx.make_range(stmt.lineno, stmt.col_offset, stmt.col_offset + len("while"))
-        return While(r, build_expr(ctx, stmt.test), [build_stmt(ctx, s) for s in stmt.body])
+        return While(r, build_expr(ctx, stmt.test),
+                     ignore_docstring([build_stmt(ctx, s) for s in stmt.body]))
 
     @staticmethod
     def build_For(ctx, stmt):
         r = ctx.make_range(stmt.lineno, stmt.col_offset, stmt.col_offset + len("for"))
         return For(
             r, [StmtBuilder.get_assign_lhs_expr(ctx, stmt.target)],
-            [build_expr(ctx, stmt.iter)], [build_stmt(ctx, s) for s in stmt.body])
+            [build_expr(ctx, stmt.iter)], ignore_docstring([build_stmt(ctx, s) for s in stmt.body]))
 
     @staticmethod
     def build_If(ctx, stmt):
         r = ctx.make_range(stmt.lineno, stmt.col_offset, stmt.col_offset + len("if"))
         return If(r, build_expr(ctx, stmt.test),
-                  [build_stmt(ctx, s) for s in stmt.body],
-                  [build_stmt(ctx, s) for s in stmt.orelse])
+                  ignore_docstring([build_stmt(ctx, s) for s in stmt.body]),
+                  ignore_docstring([build_stmt(ctx, s) for s in stmt.orelse]))
```

Comment:
I don't really like the fact that we have to put `ignore_docstring` in so many places, and it's likely we'll need to insert more filters like this one in the future. Can you please add a `build_stmts(ctx, stmts)` function that will do the list comprehension and filter out everything that's unnecessary?

Also, I think the `DocString` class is a bit unnecessary. Let's just make it return `None` to mean the absence of a meaningful statement.